lib/ex_force/oauth.ex

```diff
@@ -174,4 +231,24 @@ defmodule ExForce.OAuth do
   else
     defp hmac_fun(key, data), do: :crypto.hmac(:sha256, key, data)
   end
+
+  defp create_jwt_payload(url, payload) do
+    key = %{"pem" => Keyword.fetch!(payload, :jwt_key)}
+    signer = Joken.Signer.create("RS256", key)
```

Comment:
First of all, if I were to use this feature, I'm going to keep `signer` struct instead of PEM format to avoid extra decoding work on each request.

We may introduce an adapter for Joken - like it does with Tesla for http. That Joken adopter will tak any params `Joken.Signer.create/2` would take.

Reply:
What do you mean "extra decoding work on each request"?  In my experience, you call authenticate (e.g. via `get_token`) once to get an access token from Salesforce, which you would then use on all subsequent calls to the API for authentication.  So you only really use the pem string once, to get the access_token.  The access_token you get after JWT auth is the same as the access_token you get from any other oauth flow..

Sorry that I am having trouble following your concerns here.